### DISCUSSION TITLE
xdrip receive no data from Dexcom g6

### DISCUSSION THREAD

**Discussion by gitthiy** [NightscoutFoundation/xDrip#3418] [Q&A]
After Upgrade to "xiaomi hyperos" it looks like xdrip is not possible to receive data from Dexcom when app is in background or phone is locked.
  
Am manual click on "Datensammler neu starten" and keeping App in foregreound is only solution to receive data again.

Any configuration in advanced settings of xdrip Bluetooth connection like adjustment of watchtdog time were unsuccessfully

Attached log of xdrip ![Screenshot_2024-04-03-06-59-19-714_com.eveningoutpost.dexdrip.jpg](https://github.com/NightscoutFoundation/xDrip/assets/46824497/ca85841f-6de4-4da4-b081-99d6a8dd75ed)

Everything was running fine before update

**Comment by Navid200** [comment]
Can you post a screenshot of the status page?
 
Without that, this is all I can offer: https://navid200.github.io/xDrip/docs/Intermittent.html

**Reply by gitthiy** [reply]
Thanks for the link, I reviewed but most of advices does not exist on hyperos anymore. Neveretheless it is worth to get root cause as all supported Android phones of xiaomi and submarks will be shifted to hyperos.
But hopefully only a tick is missing.

![Screenshot_2024-04-03-07-23-57-183_com.eveningoutpost.dexdrip~2.jpg](https://github.com/NightscoutFoundation/xDrip/assets/46824497/c6a337d2-85cc-4c2c-92f7-a9fe62f091c6)

My actual workaround is to disable lock screen, run xdrip on system status, darken screen mostly 😒

**Reply by Navid200** [reply]
Xiaomi is a known problem phone because it disables apps running in the background.

Make sure you have not disabled wake workaround.

**Reply by gitthiy** [reply]
Do you have a documentation link for "wake workaround", please.

**Reply by Navid200** [reply]
Sure, it is a part of the recommended settings:  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html


To be specific, it is at:
`Settings` &#8722;> `Less common settings` &#8722;> `Other misc options` &#8722;> `Wake Workarounds`

**Reply by gitthiy** [reply]
Is it an option to take a look at the Bluetooth implementation of androidaps? This also works perfectly under hyperos. 

Just thinking out loud.

**Reply by Navid200** [reply]
androidaps?

AAPS I know is a software that does not collect from Dexcom.  xDrip can broadcast to AAPS.
What androidaps do you mean?

**Reply by gitthiy** [reply]
Exactly this androidaps. The communication of app with Insulin pump is also done by Bluetooth.

**Reply by Navid200** [reply]
A Dexcom transmitter transmits for only a few seconds once every 5 minutes.  That's how such a small battery can last for 3 months.
I doubt that the pump has such a limitation.  

This is not limited to xDrip.  Have you ever tried the build your own Dexcom app?  Do you not have similar issues with it?

**Reply by gitthiy** [reply]
Use cases are different, but integration should be same. Androidaps triggeres pump on demand, also when app is in background or when phone is locked.

hyperos identifies xdrip as a non compatible app

![Screenshot_2024-04-03-18-43-10-767_com.google.android.permissioncontroller.jpg](https://github.com/NightscoutFoundation/xDrip/assets/46824497/e6a1021c-aa19-4a45-8b7d-59f99b3ab35d)

My 5 cents to get this done.

**Reply by gitthiy** [reply]
> Sure, it is a part of the recommended settings:  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html
> 
> 
> To be specific, it is at:
> `Settings` &#8722;> `Less common settings` &#8722;> `Other misc options` &#8722;> `Wake Workarounds`

This solved my issue, in addition with switch power to Performance on hyperos.
Many thanks for supporting 👍

**Reply by gitthiy** [reply]
During last days, issue appeared again. It is very frustrating to manual trigger xdrip again and again, keeping xdrip actively on open screen. It looks like "wake workaround" works not exactly as doing this manually. 

I did have run a logcat of my device - pls see attached zip file
[log.zip](https://github.com/NightscoutFoundation/xDrip/files/14966946/log.zip)

![image](https://github.com/NightscoutFoundation/xDrip/assets/46824497/379cdbfd-c3a3-4d30-9817-e08673db2020)

![image](https://github.com/NightscoutFoundation/xDrip/assets/46824497/23eb74ef-10ae-4718-a491-76dcbc27e29b)

**Reply by Navid200** [reply]
Switch to the latest Nightly and see if it helps.

**Reply by gitthiy** [reply]
Hi Navid,
now running with latest nightly build for 24hrs. Looks pretty well.

Fingers crossing that this now solved issue entirely.

Thanks for supporting!
